miqa/core/tests/test_rest.py

```diff
@@ -96,105 +147,154 @@ def test_experiment_lock_acquire_requires_auth(api_client, experiment):
 
 
 @pytest.mark.django_db
-def test_experiment_lock_release(api_client, experiment_factory, user):
+def test_experiment_lock_release(user_api_client, experiment_factory, user):
     experiment = experiment_factory(lock_owner=user)
-    api_client.force_authenticate(user=user)
-    resp = api_client.delete(f'/api/v1/experiments/{experiment.id}/lock')
-    assert resp.status_code == 200
+    resp = user_api_client(project=experiment.project).delete(
+        f'/api/v1/experiments/{experiment.id}/lock'
+    )
+    if not any(
+        perm in Project.get_review_permission_groups()
+        for perm in get_perms(user, experiment.project)
+    ):
+        assert resp.status_code == 401
+    else:
+        assert resp.status_code == 200
 
-    experiment.refresh_from_db()
-    assert experiment.lock_owner is None
+        experiment.refresh_from_db()
+        assert experiment.lock_owner is None
 
 
 @pytest.mark.django_db
 def test_experiment_lock_only_owner_can_release(
-    authenticated_api_client, experiment_factory, user_factory
+    user_api_client, experiment_factory, user_factory, user
 ):
     owner = user_factory()
     experiment = experiment_factory(lock_owner=owner)
-    resp = authenticated_api_client.delete(f'/api/v1/experiments/{experiment.id}/lock')
-    assert resp.status_code == 409
+    resp = user_api_client(project=experiment.project).delete(
+        f'/api/v1/experiments/{experiment.id}/lock'
+    )
+    if not any(
+        perm in Project.get_review_permission_groups()
+        for perm in get_perms(user, experiment.project)
+    ):
+        assert resp.status_code == 401
+    else:
+        assert resp.status_code == 409
 
-    experiment.refresh_from_db()
-    assert experiment.lock_owner.id == owner.id
+        experiment.refresh_from_db()
+        assert experiment.lock_owner.id == owner.id
 
 
 @pytest.mark.django_db
-def test_read_without_lock_ok(authenticated_api_client, scan_decision):
-    resp = authenticated_api_client.get(f'/api/v1/scan-decisions/{scan_decision.id}')
-    assert resp.status_code == 200
+def test_read_without_lock_ok(user_api_client, scan_decision, user):
+    resp = user_api_client().get(f'/api/v1/scan-decisions/{scan_decision.id}')
+    if get_perms(user, scan_decision.scan.experiment.project) == []:
+        assert resp.status_code == 404
+    else:
+        assert resp.status_code == 200
 
 
 @pytest.mark.django_db
-def test_create_note_without_lock_fails(authenticated_api_client, scan):
-    resp = authenticated_api_client.post(
+def test_create_note_without_lock_fails(user_api_client, scan, user):
+    resp = user_api_client().post(
         '/api/v1/scan-decisions',
         data={
             'scan': scan.id,
             'decision': 'Good',
             'note': 'hello',
         },
     )
-    assert resp.status_code == 403
-    assert resp.data['detail'] == 'You must lock the experiment before performing this action.'
+    if get_perms(user, scan.experiment.project) == []:
```

Comment:
These checks would all benefit from a `has_project_permission` helper

Reply:
Agreed. I used the same helper functions in `permissions.py` that I implemented for your first suggestion.